src/device.js

```diff
@@ -219,6 +219,31 @@ export default class Device extends EventEmitter {
         return this.run(fn, {...options_, aggressive: true});
     }
 
+    getVersions(major_version: number, minor_version: number, patch_version: number, isBootloader: boolean) {
+        let result = {};
+        if (isBootloader) {
+            result = {
+                firmware_version: null,
+                bootloader_version: {
+                    major_version,
+                    minor_version,
+                    patch_version,
+                },
+            };
+        } else {
+            result = {
+                bootloader_version: null,
+                firmware_version: {
+                    major_version,
```

Comment:
maybe change firmware_varsion.major_version to firmware_version.major ? Just for convenience?